sql/truncate.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// Truncate deletes all rows from a table.
+func (p *planner) Truncate(n *parser.Truncate) (planNode, error) {
+	b := client.Batch{}
```

Comment:
I see sql-level transactions and kv transactions mixed up in all those issues, I don't think about sql transactions (yet) but only about updating atomically on the KV level. Probably makes sense if each `planNode` returned contains a batch for now and the planner executes it all in a KV transaction at the end. It won't quite be enough because that doesn't allow rearranging the query tree and streaming results one by one, but it's conceptually closer.